test/helpers/vertical_nav_helper_test.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class VerticalNavHelperTest < ActionView::TestCase
+
+  test '#backend_api_nav_sections' do
+    @backend_api = FactoryBot.create(:backend_api)
+
+    # if permitted
+    stubs(can?: true)
+    assert_equal(["Overview", "Analytics", "Methods & Metrics", "Mapping Rules"], backend_api_nav_sections.pluck(:title))
+
+    # if not permitted
+    stubs(can?: false)
+    assert_equal(["Overview", "Methods & Metrics", "Mapping Rules"], backend_api_nav_sections.pluck(:title))
+
+    # When backend_api is not persisted
+    @backend_api = BackendApi.new
+    assert_equal([], backend_api_nav_sections.pluck(:title))
+
+
+    # When backend_api is nil
+    @backend_api = nil
+    assert_equal([], backend_api_nav_sections.pluck(:title))
```

Comment:
You do not have to use pluck(:title) when you know it is an empty Hash btw ;)
That is fine for now